src/Sulu/Bundle/WebsiteBundle/Routing/PortalLoader.php

```diff
@@ -47,7 +47,7 @@ public function load($resource, $type = null)
                     $importedRoute->getDefaults(),
                     array_merge(['prefix' => '.*', 'host' => '.+'], $importedRoute->getRequirements()),
                     $importedRoute->getOptions(),
-                    (!empty($importedRoute->getHost()) ? $importedRoute->getHost() : '{host}'),
```

Comment:
@wachterjohannes You have introduced that line in #2384, do you remember why? From what I can see `{host}` would have matched any route anyway, so it should be the same as if you put no host there at all. Or am I overlooking something?

Reply:
I personally would prefer removing here `{host}` and set it to empty so it uses `ANY` instead of `{host}` for this routes:

![Bildschirmfoto 2020-02-05 um 08 59 23](https://user-images.githubusercontent.com/1698337/73822223-e901f280-47f5-11ea-82e9-5bc804784c74.png)

Reply:
ping @wachterjohannes

Reply:
That's what I have basically done here, right? :see_no_evil:

Reply:
Oh sorry did look at the old code 🙈 :

Bad sadly I tested this and I think because of bc we can not remove the `host` as if we remove it it will add the one which generated routes adding the host as queryparameter e.g. `/search?host=test.com`

so I think we should go with setting the host to empty as default e.g.:

```php
            $defaults = $importedRoute->getDefaults();
            if (empty($importedRoute->getHost())) {
                $defaults['host'] = '';
            }

            $collection->add(
                $importedRouteName,
                new PortalRoute(
                    '{prefix}' . $importedRoute->getPath(),
                    $defaults,
                    array_merge(['prefix' => '.*', 'host' => '.+|^$'], $importedRoute->getRequirements()),
                    $importedRoute->getOptions(),
                    !empty($importedRoute->getHost()) ? $importedRoute->getHost() : '{host}',
                    $importedRoute->getSchemes(),
                    $importedRoute->getMethods(),
                    $condition . (!empty($importedCondition) ? ' and (' . $importedCondition . ')' : '')
                )
            );
```

What do you think?

Reply:
@danrot ok that doesn't work as expected maybe you  have an idea I'm trying the following:

```yaml
# config/services.xml
parameters:
    router.request_context.host: 'example.org:8181'
```

```php
<?php

// src/Command/RouteTestCommand.php

namespace App\Command;

use Symfony\Component\Console\Command\Command;
use Symfony\Component\Console\Input\InputInterface;
use Symfony\Component\Console\Output\OutputInterface;
use Symfony\Component\Console\Style\SymfonyStyle;
use Symfony\Component\Routing\RouterInterface;

class RouteTestCommand extends Command
{
    /**
     * @var RouterInterface
     */
    private $router;

    /**
     * RouteTestCommand constructor.
     * @param RouterInterface $router
     */
    public function __construct(RouterInterface $router)
    {
        $this->router = $router;

        parent::__construct('app:test:route');
    }

    protected function execute(InputInterface $input, OutputInterface $output)
    {
        $routeOnlyPrefix = $this->router->generate(
            'sulu_search.website_search',
            [
                'prefix' => '/en',
            ],
            0
        );
        $expectedRouteOnlyPrefix = 'http://example.org:8181/en/search';

        $routeWithCustomHost = $this->router->generate(
            'sulu_search.website_search',
            [
                'prefix' => '/en',
                'host' => 'mytestdomain.com',
            ],
            0
        );
        $expectedRouteWithCustomHost = 'http://mytestdomain.com/en/search';

        $symfonyRoute = $this->router->generate('sulu_website.sitemap_index', [], 0);
        $expectedSymfonyRoute = 'http://example.org:8181/sitemap.xml';

        $data = [
            [
                'generatedRoute' => $routeOnlyPrefix,
                'expectedRoute' => $expectedRouteOnlyPrefix,
                'match' => $routeOnlyPrefix === $expectedRouteOnlyPrefix,
            ],
            [
                'generatedRoute' => $routeWithCustomHost,
                'expectedRoute' => $expectedRouteWithCustomHost,
                'match' => $routeWithCustomHost === $expectedRouteWithCustomHost,
            ],
            [
                'generatedRoute' => $symfonyRoute,
                'expectedRoute' => $expectedSymfonyRoute,
                'match' => $symfonyRoute === $expectedSymfonyRoute,
            ],
        ];

        $ui = new SymfonyStyle($input, $output);

        $ui->table(array_keys($data[0]), $data);
    }
}
```

Can not get all cases working as expected:

![Bildschirmfoto 2020-02-05 um 11 05 18](https://user-images.githubusercontent.com/1698337/73831956-86195700-4807-11ea-99fe-5384da2cd505.png)

![Bildschirmfoto 2020-02-05 um 11 05 53](https://user-images.githubusercontent.com/1698337/73831960-87e31a80-4807-11ea-81f3-69fb41134bfd.png)

Reply:
@danrot but If I think about it if you have use `{host}` in dev env and used as recommended the real domain in prod the same would had happened on prod. So maybe we should use your change and never set the `{host}` here.